### DISCUSSION TITLE
Only first call of readContracts returns data?

### DISCUSSION THREAD

**Discussion by thibauld** [wevm/wagmi#1777] [Q&A] [answered]
Hi guys :wave: 

We're currently looking at using wagmi at Fairmint because it looks like you guys did a great job.
However, we're a bit puzzled when doing our initial tests. See the screenshot below :point_down: 
![image](https://user-images.githubusercontent.com/197119/216365793-96cb1baa-95eb-414d-99ea-9c6add0ad23f.png)

How come data is only retrieved in the first call? 
We're probably missing something here... but I can't see what.
Any hint would be very helpful, thanks!

Here is the test code:
```
import { ethers } from "ethers";
import {
  createClient, readContracts
} from "@wagmi/core";
import abi from "./src/cafe-abi.js";

async function getData() {
  try {
    const nodeUrl = "https://eth-mainnet.alchemyapi.io/v2/xxxxx";
    const provider = new ethers.providers.JsonRpcProvider(nodeUrl);
    createClient({
      provider
    });

    const datContract = {
      address : "0x3983c6da8235f59e16a034ee773e60a69e8f14b2",
      abi     : abi.dat,
    };
    const contracts = [
      {
        ...datContract,
        functionName: "currency",
      },
    ];

    const data1 = await readContracts({ contracts });
    console.log({ data1 });

    const data2 = await readContracts({ contracts });
    console.log({ data2 });

    const data3 = await readContracts({ contracts });
    console.log({ data3 });
  }
  catch (err) {
    console.log(err);
  }
}

getData();
```

**Comment by tmm** [comment]
Can you post a full example? What ABI are you using?

**Reply by tmm** [reply]
Appreciate you providing this, but it's still far from a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) I can use to take a look.

**Reply by tmm** [reply]
Thought it was just the ABI before. Can you post the code in a repo on GitHub? Download zips from the Internet isn’t something I love doing.

**Comment by thibauld** [comment, marked as answer]
UPDATE: we found the issue. We needed to use wagmi provider: https://wagmi.sh/core/providers/alchemy
After that, everything worked as expected :ok_hand:
